opencensus/trace/trace_options.h

```diff
@@ -42,6 +42,10 @@ class TraceOptions final {
   // the process, e.g. to Stackdriver/Zipkin.
   bool IsSampled() const;
 
+  // Set IsSampled for the TraceOptions. Normally, this is invoked
+  // implicitly by StartSpan
+  void SetSampled(bool is_sampled);
```

Comment:
Please don't do this. We want the option to be immutable. If you want a helper function add a static function:

TraceOption SetSampled(TraceOption parent)

Reply:
Are you proposing a static function that takes just the TraceOption argument like
`TraceOption SetSampled(TraceOption parent)`
or an additional argument like
`TraceOption SetSampled(TraceOption parent, bool isSampled)`

Both of which will make a new copy, but the latter option allows an additional input. Thanks

Reply:
Fine with the second option. Just want to make sure that the object is immutable. This way we don't need to do any synchronization, also once a Span is created with a TraceOption then it will be the same for the entire life.

Reply:
Ack.

That makes sense. Thanks, I will implement the second option then.

Reply:
updated. 

I think the CI failure is not related to my change though?
https://travis-ci.org/census-instrumentation/opencensus-cpp/builds/495589423?utm_source=github_status&utm_medium=notification

Reply:
Currently:
```
TraceOptions b = TraceOptions::SetSampled(a, false);
```

Can we change the name? Also, it doesn't need to be static. How about:
```
TraceOptions b = a.WithSampling(true);
```

Reply:
(CI failure was fixed in #282)